pkg/addon-operator/operator.go

```diff
@@ -1481,6 +1481,17 @@ func (op *AddonOperator) HandleModuleRun(t sh_task.Task, labels map[string]strin
 				logEntry.Debugf("ModuleRun '%s' phase", baseModule.GetPhase())
 
 				treg := trace.StartRegion(context.Background(), "ModuleRun-OnStartup")
+				defer treg.End()
+
+				// Ensure modules CRDs
+				if err := op.EnsureCRDs(baseModule); err != nil {
+					op.ModuleManager.UpdateModuleLastErrorAndNotify(baseModule, err)
+					res.Status = queue.Fail
```

Comment:
we can have different types of errors, conflict, or apiserver unavailability, or validation errors. Since it is difficult to determine what type of error it is, we mark the task as erroneous and repeat it until the problem is solved. Yes, this leads to the fact that in case of problems with crd validation we will repeat the operation infinitely